repos/system_upgrade/el7toel8/libraries/repository_scanning.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+from leapp.libraries.common import mounting, utils
+from leapp.models import RepositoryFile, RepositoryData, fields
+
+
+def _parse_repository(repoid, repo_data):
+    def asbool(x):
+        return x == '1'
+    prepared = {'repoid': repoid, 'additional_fields': {}}
+    for key in repo_data.keys():
+        if key in RepositoryData.fields:
+            if isinstance(RepositoryData.fields[key], fields.Boolean):
+                repo_data[key] = asbool(repo_data[key])
+            prepared[key] = repo_data[key]
+        else:
+            prepared['additional_fields'][key] = repo_data[key]
+    prepared['additional_fields'] = json.dumps(prepared['additional_fields'])
+    return RepositoryData(**prepared)
+
+
+def parse_repofile(repofile):
+    """
+    Parse the given repo file.
+
+    :param repofile: Path to the repo file
+    :type repofile: str
+    :rtype: RepositoryFile
+    """
+    data = []
+    with open(repofile, mode='r') as fp:
+        cp = utils.parse_config(fp)
+        for repoid in cp.sections():
+            data.append(_parse_repository(repoid, dict(cp.items(repoid))))
+    return RepositoryFile(file=repofile, data=data)
+
+
+def scan_repositories(context=mounting.NotIsolatedActions(base_dir='/')):
```

Comment:
```suggestion
def get_repofiles(context=mounting.NotIsolatedActions(base_dir='/')):
```
Or `get_parsed_repofiles` if you want emphasize the fact that what you return are objects of the parsed repofiles.